protocol/backend/backend.go

```diff
@@ -237,7 +238,31 @@ func (b *backend) ClearXMRDepositAddress(offerID types.Hash) {
 }
 
 // HandleRelayClaimRequest validates and sends the transaction for a relay claim request
-func (b *backend) HandleRelayClaimRequest(request *message.RelayClaimRequest) (*message.RelayClaimResponse, error) {
+func (b *backend) HandleRelayClaimRequest(
+	remotePeer peer.ID,
+	request *message.RelayClaimRequest,
+) (*message.RelayClaimResponse, error) {
+	if request.OfferID != nil {
+		has := b.swapManager.HasOngoingSwap(*request.OfferID)
+		if !has {
+			return nil, fmt.Errorf("cannot relay taker-specific claim request; no ongoing swap for swap %s", *request.OfferID)
+		}
+
+		info, err := b.swapManager.GetOngoingSwap(*request.OfferID)
+		if err != nil {
+			return nil, err
+		}
+
+		if info.Provides != coins.ProvidesETH {
```

Comment:
Maybe we need `info.IsTaker()` and `info.IsMaker()` methods that does these checks? (Shorter and more obvious than these checks which are easy confused with eth-asset checks.) If you like the idea, I can implement it in a future PR.

Reply:
I added an `info.IsTaker()` method for this